src/frontend/src/binder/expr/function.rs

```diff
@@ -219,35 +229,59 @@ impl Binder {
 
     pub(super) fn bind_window_function(
         &mut self,
+        kind: WindowFuncKind,
+        inputs: Vec<ExprImpl>,
         WindowSpec {
             partition_by,
             order_by,
             window_frame,
         }: WindowSpec,
-        function_name: String,
-        inputs: Vec<ExprImpl>,
     ) -> Result<ExprImpl> {
         self.ensure_window_function_allowed()?;
-        if let Some(window_frame) = window_frame {
-            return Err(ErrorCode::NotImplemented(
-                format!("window frame: {}", window_frame),
-                None.into(),
-            )
-            .into());
-        }
-        let window_function_type = WindowFunctionType::from_str(&function_name)?;
         let partition_by = partition_by
             .into_iter()
             .map(|arg| self.bind_expr(arg))
             .try_collect()?;
-
         let order_by = OrderBy::new(
             order_by
                 .into_iter()
                 .map(|order_by_expr| self.bind_order_by_expr(order_by_expr))
                 .collect::<Result<_>>()?,
         );
-        Ok(WindowFunction::new(window_function_type, partition_by, order_by, inputs)?.into())
+        let frame = if let Some(frame) = window_frame {
+            Some(match frame.units {
+                WindowFrameUnits::Rows => {
+                    let convert_bound = |bound| match bound {
+                        WindowFrameBound::CurrentRow => FrameBound::CurrentRow,
+                        WindowFrameBound::Preceding(None) => FrameBound::UnboundedPreceding,
+                        WindowFrameBound::Preceding(Some(offset)) => {
+                            FrameBound::Preceding(offset as usize)
+                        }
+                        WindowFrameBound::Following(None) => FrameBound::UnboundedFollowing,
+                        WindowFrameBound::Following(Some(offset)) => {
+                            FrameBound::Following(offset as usize)
+                        }
+                    };
+                    let start = convert_bound(frame.start_bound);
+                    let end = if let Some(end_bound) = frame.end_bound {
+                        convert_bound(end_bound)
+                    } else {
+                        FrameBound::CurrentRow
+                    };
+                    Frame::Rows(start, end)
```

Comment:
Shall we do some checks before constructing this frame?

> https://www.postgresql.org/docs/current/sql-expressions.html#SYNTAX-WINDOW-FUNCTIONS
>
> where frame_start and frame_end can be one of
> ```
> UNBOUNDED PRECEDING
> offset PRECEDING
> CURRENT ROW
> offset FOLLOWING
> UNBOUNDED FOLLOWING
> ```
> 
> Restrictions are that frame_start cannot be UNBOUNDED FOLLOWING, frame_end cannot be UNBOUNDED PRECEDING, and the frame_end choice cannot appear earlier in the above list of frame_start and frame_end options than the frame_start choice does — for example RANGE BETWEEN CURRENT ROW AND offset PRECEDING is not allowed. But, for example, ROWS BETWEEN 7 PRECEDING AND 8 PRECEDING is allowed, even though it would never select any rows.